lib/IRGen/IRGen.cpp

```diff
@@ -490,10 +490,13 @@ static createTargetMachine(IRGenOptions &Opts, ASTContext &Ctx) {
   }
 
   // Create a target machine.
-  llvm::TargetMachine *TargetMachine
-    = Target->createTargetMachine(Triple.str(), CPU,
-                                  targetFeatures, TargetOpts, Reloc::PIC_,
-                                  CodeModel::Default, OptLevel);
+  auto cmodel = CodeModel::Default;
+  if (Triple.isWindowsCygwinEnvironment())
+    cmodel = CodeModel::Large;
```

Comment:
Actually, thinking more about it, do you have the code from an instance where this happens?  The code generation issue that you sounds very much like a bug in LLVM.  x86 has fully addressability, so within a 32-bit range should be fine.